### DISCUSSION TITLE
dinamic window gaps

### DISCUSSION THREAD

**Discussion by voziand** [bakkeby/dwm-flexipatch#290] [How To]
So i tried to patch dwm with bar-padding, systray, fullgaps by myself, it all worked well until i tried to apply the status2d patch. After applying that patch i was getting the status bar showing twice, tried to fix it but no success. So i decided to give Flexipatch a go.
i applied the following patches with flexipatch:

#define BAR_STATUS2D_PATCH 1
#define BAR_HEIGHT_PATCH 1
#define BAR_LTSYMBOL_PATCH 1
#define BAR_STATUS_PATCH 1
#define BAR_SYSTRAY_PATCH 1
#define BAR_TAGS_PATCH 1
#define BAR_WINTITLE_PATCH 1
#define BAR_TITLE_LEFT_PAD_PATCH 1
#define BAR_PADDING_PATCH 1
#define TILE_LAYOUT 1
#define MONOCLE_LAYOUT 1
#define VANITYGAPS_PATCH 1

it all worked well except i can't change dynamically the gaps between windows anymore.
i noticed that the dwm version i have where i manually patched it, has these function in the config.def.h

	{ MODKEY,                       XK_minus,  setgaps,        {.i = -1 } },
	{ MODKEY,                       XK_equal,  setgaps,        {.i = +1 } },
	{ MODKEY|ShiftMask,             XK_equal,  setgaps,        {.i = 0  } },

is there anyway to get this functionality with flexypatch?
let me know if i can provide any more details.

if it's something that requires a lot of work, is anyone up to maybe help me fix the dwm build where i manually patched?

Also, thanks fro this project! it's awesome.

**Comment by bakkeby** [comment]
Vanitygaps allows you to control the horizontal and vertical gaps independently, and outer and inner gaps independently. 

Here are the default keybindings:
https://github.com/bakkeby/dwm-flexipatch/blob/b93a5de5587432f21a850c2e86f8478433c30c8e/config.def.h#L988-L1003

The first two should be equivalent to the `setgaps` function you had before.

**Reply by voziand** [reply]
cool, thanks.. 
helps if i read.. 

on another note, i'm having this weird issue where icons in the system tray have these corners around them.
![image](https://user-images.githubusercontent.com/60975283/185720823-4a126d4a-02e9-4140-8af9-955408b98c53.png)

would this be an issue with the systray or is it picom that is messing up things?

**Reply by voziand** [reply]
it was picom.
